src/sinks/sftp/test.rs

```diff
@@ -0,0 +1,93 @@
+use bytes::Bytes;
+use codecs::{encoding::Framer, JsonSerializerConfig, NewlineDelimitedEncoderConfig};
+use vector_common::request_metadata::GroupedCountByteSize;
+use vector_core::partition::Partitioner;
+
+use super::config::SftpConfig;
+use crate::{
+    codecs::{Encoder, EncodingConfigWithFraming, SinkType},
+    event::LogEvent,
+    sinks::{
+        opendal_common::{OpenDalRequest, OpenDalRequestBuilder},
+        util::{
+            request_builder::{EncodeResult, RequestBuilder},
+            Compression,
+        },
+    },
+};
+
+fn default_config(encoding: EncodingConfigWithFraming) -> SftpConfig {
+    SftpConfig {
+        root: "/tmp/".to_string(),
+        prefix: "%F/".to_string(),
+        endpoint: "127.0.0.1:22".to_string(),
+        user: "ubuntu".to_string(),
+        key: "/home/ubuntu/.ssh/id_rsa".to_string(),
+        known_hosts_strategy: "strict".to_string(),
+        encoding,
+        compression: Compression::gzip_default(),
+        batch: Default::default(),
+        acknowledgements: Default::default(),
+    }
+}
+
+#[test]
+fn Sftp_generate_config() {
+    crate::test_util::test_generate_config::<SftpConfig>();
+}
+
+fn request_builder(sink_config: &SftpConfig) -> OpenDalRequestBuilder {
+    let transformer = sink_config.encoding.transformer();
+    let (framer, serializer) = sink_config
+        .encoding
+        .build(SinkType::MessageBased)
+        .expect("encoding must build with success");
+    let encoder = Encoder::<Framer>::new(framer, serializer);
+
+    OpenDalRequestBuilder {
+        encoder: (transformer, encoder),
+        compression: sink_config.compression,
+    }
+}
```

Comment:
Rather than re-implement this logic here, we should extract the corresponding logic out of `SftpConfig::build_processor` into a separate method so that we can test it in isolation.